Integrations/SignalSciences/SignalSciences.py

```diff
@@ -0,0 +1,1001 @@
+    if '.' in string_var:
+        return False
+    if string_var[0] in ('-', '+'):
+        return string_var[1:].isdigit()
+    return string_var.isdigit()
+
+
+def is_legal_interval_for_alert(interval):
+    if not represents_int(interval):
+        return False
+    interval_int = int(interval)
+    if not (interval_int == 1 or interval_int == 10 or interval_int == 60):
+        return False
+    return True
+
+
+# the validity of the tagName argument is currently unchecked, until further information is provided
+def validate_alert_args(siteName, long_name, tag_name, interval, threshold, enabled, action):
+    if not represents_int(threshold):
+        return_error("Error: {0} is not a valid threshold value. Threshold must be an integer".format(threshold))
+    if not is_legal_interval_for_alert(interval):
+        return_error("Error: {0} is not a valid interval value. Interval value must be 1, 10 or 60".format(interval))
+    if len(long_name) < 3 or len(long_name) > 25:
+        return_error("Error: Illegal value for long_name argument - long_name must be between 3 and 25 characters long")
+    if not (enabled == 'True' or enabled == 'False'):
+        return_error("Error: Illegal value for 'enabled' argument - value must be 'True' or 'False'")
+    if not (action == 'info' or action == 'flagged'):
+        return_error("Error: Illegal value for 'action' argument - value must be 'info' or 'flagged'")
+
+
+def list_entry_context_from_response(response_data):
+    entry_context = {
+        'Name': response_data.get('name', ''),
+        'Type': response_data.get('type', ''),
+        'Entries': response_data.get('entries', ''),
+        'ID': response_data.get('id', ''),
+        'Description': response_data.get('description', ''),
+        'CreatedBy': response_data.get('createdBy', ''),
+        'CreatedDate': response_data.get('created', ''),
+        'UpdatedDate': response_data.get('updated', '')
+    }
+    return entry_context
+
+
+# where do skip notifications come from
+def alert_entry_context_from_response(response_data):
+    entry_context = {
+        'ID': response_data.get('id', ''),
+        'SiteID': response_data.get('siteId', ''),
+        'TagName': response_data.get('tagName', ''),
+        'LongName': response_data.get('longName', ''),
+        'Interval': response_data.get('interval', ''),
+        'Threshold': response_data.get('threshold', ''),
+        'BlockDurationSeconds': response_data.get('blockDurationSeconds', ''),
+        'SkipNotifications': response_data.get('skipNotifications', ''),
+        'Enabled': response_data.get('enabled', ''),
+        'Action': response_data.get('action', ''),
+        'CreatedDate': response_data.get('created', ''),
+    }
+    return entry_context
+
+
+def check_ip_is_valid(ip):
+    if not is_ip_valid(ip):
+        return_error("Error: IP argument is invalid. Please enter a valid IP address")
+
+
+def format_update_list_entries(entries_list, method):
```

Comment:
not a good name, maybe `generate_data_for_update_list_request`

Reply:
Why not to return something in with the 
`
{
        'entries': entries_in_update_format
}`
format

Reply:
because I think the entire data dict should be built, at least conceptually, in the same method